debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbOffsetContext.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.bson.Document;
+
+import io.debezium.connector.SnapshotRecord;
+import io.debezium.pipeline.spi.OffsetContext;
+import io.debezium.pipeline.txmetadata.TransactionContext;
+import io.debezium.schema.DataCollectionId;
+
+/**
+ * A context that facilitates the management of the current offsets across a set of mongodb replica sets.
+ *
+ * @author Chris Cranford
+ */
+public class MongoDbOffsetContext implements OffsetContext {
+
+    private final SourceInfo sourceInfo;
+    private final TransactionContext transactionContext;
+    private final Map<ReplicaSet, ReplicaSetOffsetContext> replicaSetOffsetContexts = new ConcurrentHashMap<>();
+
+    public MongoDbOffsetContext(SourceInfo sourceInfo, TransactionContext transactionContext) {
+        this.sourceInfo = sourceInfo;
+        this.transactionContext = transactionContext;
+    }
+
+    public MongoDbOffsetContext(SourceInfo sourceInfo, TransactionContext transactionContext, Map<ReplicaSet, Document> offsets) {
+        this(sourceInfo, transactionContext);
+        offsets.forEach((replicaSet, document) -> sourceInfo.opLogEvent(replicaSet.replicaSetName(), document, document, 0));
+    }
+
+    void startInitialSync(String replicaSetName) {
```

Comment:
Could we use {{snapshot}} instead of {{initialSync}} to be consistent with other connectors? Same for other instances

Reply:
Absolutely.

Reply:
See commit 0158b56.